### ISSUE TITLE
The published date is sometimes strangely formatted

### ISSUE THREAD

**Issue by Elcoid** [jeff-hughes/shellcaster#36]
The issue doesn't seem to be easily reproducible, but I was able to see it a few times. The date is sometimes formatted as in the screenshot below, with a space missing between the month and the day, and an extra 1 after the year:
![ksnip_20210721-145758](https://user-images.githubusercontent.com/47006046/126546550-2170bbb5-42b0-4aea-b303-a152d83a2804.png)

This only seems to occur when the list of episodes is scrolled down. When the problematic episodes are displayed when the list is scrolled up, the date is correctly displayed. See these two videos for example:

https://user-images.githubusercontent.com/47006046/126553076-b49d695c-d01b-41b9-bc58-64f8042a7a63.mp4

https://user-images.githubusercontent.com/47006046/126553090-84331249-825d-4219-873a-e98b7ca43577.mp4

**Comment by jeff-hughes** [comment]
I don't seem to get any actual video from those files, unfortunately. Not sure what's happening there. Is it possible for you to upload them elsewhere and link to them?

At any rate....hmm, that's not a bug I've experienced, and I don't think I have a good theory for how it could be happening. It's a little odd if it's only happening when scrolling down but not up... 

Could you let me know:
- Your OS/distro
- What terminal emulator you're using
- What version of shellcaster you're using (and if you're using the development branch or the master branch -- or if you're using the available AUR packages, if you're using the "shellcaster-git" package or the "shellcaster"/"shellcaster-bin" package)

**Comment by Elcoid** [comment]
Sure, here are links. For some reason however, I'm not able to watch them directly in Firefox. But I can download them and watch them with vlc or mpv.

[This Week in Linux](https://tuxdigital.com/thisweekinlinux/), episode 141:
https://user-images.githubusercontent.com/47006046/126553090-84331249-825d-4219-873a-e98b7ca43577.mp4

[Darknet Diaries](https://darknetdiaries.com/), episodes 82-84:
https://user-images.githubusercontent.com/47006046/126553076-b49d695c-d01b-41b9-bc58-64f8042a7a63.mp4

I'm on Artix Linux with Konsole 21.04.3 and shellcaster-bin 1.2.0-1 installed from the AUR.

**Comment by jeff-hughes** [comment]
Thanks for that -- same thing happens to me, the video didn't show up in Firefox, but downloading them worked. Weird.

Anyway, I added the two feeds you shared, and using the latest release of shellcaster I was able to replicate what you were seeing. My best guess at this point is that it seems to be ncurses not always properly refreshing the line of text for some reason. Not sure exactly why that's happening -- more likely to be something I've done wrong given that ncurses is pretty stable at this point. But it looks like if the previous line was longer, sometimes writing a shorter line will leave some artifacts. That might not be the full story -- I also saw things like "January117" instead of "January 17", so that extra 1 instead of a space is coming from somewhere. Same thing with your screenshot where "March7" is missing the space. But I'd really have to do some more extensive testing since it seems a little inconsistent.

However, the development branch of shellcaster does seem to fix the issue. In that branch I've completely pulled out ncurses entirely and am using crossterm (a Rust-native terminal library) instead. I don't currently _recommend_ using the development branch as it's still a work in progress; there are definitely some bugs to work out there as well that may be more annoying to you than oddly formatted dates :) But either way, it appears that crossterm is doing the job properly, so the next release of shellcaster should fix the issue. I'm still wanting to add some more features before release, but that release will be coming.......sometime. Whenever I stop procrastinating!
